apps/internal/base/base.go

```diff
@@ -280,26 +280,8 @@ func (b Client) AuthCodeURL(ctx context.Context, clientID, redirectURI string, s
 }
 
 func (b Client) AcquireTokenSilent(ctx context.Context, silent AcquireTokenSilentParameters) (AuthResult, error) {
+	// when tenant == "", the caller didn't specify a tenant and WithTenant will use the client's configured tenant
 	tenant := silent.TenantID
-	if tenant == "" {
-		// the caller didn't specify a tenant, so we'll use the client's configured tenant or the given account's home tenant
-		switch tenant = b.AuthParams.AuthorityInfo.Tenant; tenant {
```

Comment:
The PR looks surprisingly simple. Is it because the MSAL Go was indeed largely written in the tenant-agnostic way previously, so that this new PR merely needs to revert those logic that we introduced in the recent PR #366?

One question here. If caller did not specify a tenant, don't we still need `tenant = b.AuthParams.AuthorityInfo.Tenant`?

Reply:
`tenant` is used only as the argument to `WithTenant()`. When it's `""`, that method sets the tenant for this authentication to the client's default, which is carried by `b.AuthParams`:

https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/8531f217b6e983184cab9bba88b80600ea879d2b/apps/internal/oauth/ops/authority/authority.go#L190-L194

It does appear that MSAL Go was tenant agnostic before #366; I suppose the relevant parts of the implementation were inspired by MSAL Python? The dynamic tenant feature added the first special handling of "common" and "organizations".